### PR TITLE
Scoreboard app

### PR DESCRIPTION
**Pull request by gjbianco**
screenshot

### CODE REVIEW

**Comment by jramcast** [comment]
My tests:

## RHEL

- RHEL CSB 8.0
- node v14.14.0
- npm 6.14.8
- `cy:run` :heavy_check_mark: 
- `cy:docker`: :x:
  - Cypress is not able to connect to the specified url (http://host.docker.internal:3000).
  - Changing the base url to http://172.17.0.1:3000 did not fix the problem.
- `cy:podman` 
  - Cypress is not able to connect to the specified url (http://host.docker.internal:3000). :x:
  - Working after changing the baseUrl to http://172.17.0.1:3000 (`"cy:podman": "podman run -it --rm -v $PWD:/e2e -w /e2e cypress/included:5.4.0 --config baseUrl=http://172.17.0.1:3000"`)  :heavy_check_mark:  
- `cy:open`:
  - Chrome 86 :heavy_check_mark: 
  - Chromium 85 :heavy_check_mark: 
  - Firefox 68 :heavy_check_mark: 

## macOS

- macOS Catalina 10.15.7
- node v14.2.0
- npm 6.14.4
- `cy:run` :heavy_check_mark: 
- `cy:docker`: :heavy_check_mark: 
- `cy:open`:
  - Chrome 86 :heavy_check_mark: 
  - Firefox 78 :heavy_check_mark:

**Comment by jramcast** [comment]
If this is the solution, we should be merging to the `solutions` branch

**Comment by psolarvi** [comment]
In my case:

### RHEL CSB 8.0
I was unable to install the project dependencies using only what is available in the yum repos. My steps were:
```
[user@host]$ sudo dnf install -y nodejs
[user@host]$ sudo npm install -g yarn
[user@host]$ cd .../scoreboard
[user@host]$ yarn install
...
> error @jest/types@26.6.0: The engine "node" is incompatible with this module. Expected version ">= 10.14.2". Got "10.14.1"
```

Node version: 
```
[user@host]$ node --version
v10.14.1
```

**Comment by gjbianco** [comment]
> If this is the solution, we should be merging to the `solutions` branch

This was for a GE, so there wouldn't really be a "non-solution" version, right?

**Comment by gjbianco** [comment]
@psolarvi yeah, that's a pretty old version of Node.js at this point ☹️ https://nodejs.org/en/about/releases/

It's an LTS release and looks like it'll go completely out of maintenance not long after this course should be released, iirc. We'll only see more incompatibility issues with v10 as time goes on (as packages drop support for it).

We probably need to discuss how students should be installing Node.js versions if RHEL is shipping with (nearly) outdated versions. One possible solution is to have them use NVM: https://github.com/nvm-sh/nvm

**Comment by gjbianco** [comment]
@jramcast wow, that list is awesome! Thanks for doing all that! I can add the compatibility changes you mentioned and we can discuss further tweaks to try and improve it on other platforms.

**Comment by psolarvi** [comment]
I just checked in my other machine and RHEL 8.2 (up to date) has node version `10.21.0`

**Comment by gjbianco** [comment]
> I just checked in my other machine and RHEL 8.2 (up to date) has node version `10.21.0`

So I'm guessing it didn't give you that same error?

Still a bit concerning IMO since, again, v10 is unsupported after April. Probably not a lot we can do about it, either way.

File: scoreboard/cypress/fixtures/example.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
+  "body": "Fixtures are a great way to mock data for responses to routes"
+}
```
Comment (jramcast) [inline scoreboard/cypress/fixtures/example.json:5]:
```suggestion
}

```

**Review by jramcast — APPROVED** [review]
All good

File: scoreboard/cypress/fixtures/example.json (hunk 1 above)
Comment (jramcast) [reply scoreboard/cypress/fixtures/example.json:5]: Not necessary for this PR, but we could update the CI file https://github.com/RedHatTraining/DO400-apps/blob/solutions/Jenkinsfile-curriculum to run the dockerized functional tests.

**Pull request merged**